packages/web/src/components/admin_page/feature_switch_page/render_switch.jsx

```diff
@@ -0,0 +1,178 @@
+            <Form.Group className="mb-3" controlId="formBasicEmail">
+                <Form.Label>Set a feature:</Form.Label>
+                <Form.Control
+                    type="text"
+                    placeholder="Enter new feature switch"
+                    onChange={handleChange}
+                    name="switch_id"
+                    value={newSwitch.switch_id}
+                />
+                <Form.Text className="text-muted">Add a new feature</Form.Text>
+            </Form.Group>
+            <Form.Group className="mb-3" controlId="formBasicIsEnabled">
+                <Form.Label>Set it to true or false:</Form.Label>
+                <Form.Select onChange={handleChange} name="is_enabled" value={newSwitch.is_enabled.toString()}>
+                    <option value="true">True</option>
+                    <option value="false">False</option>
+                </Form.Select>
+                <Form.Text className="text-muted"></Form.Text>
+            </Form.Group>
+
+            <Button type="submit" variant="primary">
+                Add
+            </Button>
+        </Form>
+    );
+};
+
+export const popover = (onUpdate, errorMessage) => {
+    console.log(errorMessage);
+    return (
+        <Popover id="popover-basic">
+            <Popover.Header as="h3">Want to update or remove?</Popover.Header>
+            <Popover.Body>
+                {errorMessage ? <Alert variant="warning">{errorMessage}</Alert> : null}
+                <UpdateSwitch onUpdate={onUpdate} />
+            </Popover.Body>
+        </Popover>
+    );
+};
+
+// Creates a new object updateSwitches that makes a copy of switches and updates a specific switch's is_enabled property
+const updateSwitches = (switches, setSwitches) => data => {
+    setSwitches({ ...switches, [data.switch_id]: data.is_enabled });
+};
+
+//creates a new object updatedSwitches that is equal to switches with the siwtchToRemove, removed
+const deleteSwitches = (switches, setSwitches) => switchToRemove => {
+    // eslint-disable-next-line no-unused-vars
+    const { [switchToRemove]: _, ...updatedSwitches } = switches;
+    setSwitches(updatedSwitches);
+};
+
+const RenderSwitches = () => {
+    //We need to pass what is selected to the child element to track if a switch is selected so that we dont display two messages for editing and adding switch
+    const [selected, setSelected] = useState('');
+    const [showAddOverlay, setShowAddOverlay] = useState(false);
+    const [featureSwitches, setFeatureSwitches] = useContext(FsContext);
+    const [errorMessage, setErrorMessage] = useState(undefined);
+
+    // Create functions to update and delete feature switches and call it by the handleUpdateSwitches/handleDeleteSwitches
+    const handleUpdateSwitches = updateSwitches(featureSwitches, setFeatureSwitches);
+    const handleDeleteSwitches = deleteSwitches(featureSwitches, setFeatureSwitches);
+
+    //Check if the API is working
+    useEffect(() => {
```

Comment:
I see where the confusion is coming from. 


The `API IS NOT RESPONDING` message should be coming only from POST requests. We don't check on initial load whether we were able to fetch the switches, but only when we try to create a new one or update an old one.